libgit2-sys/lib.rs

```diff
@@ -3551,7 +3553,21 @@ pub fn init() {
         openssl_init();
         ssh_init();
         let r = git_libgit2_init();
-        assert!(r >= 0, "couldn't initialize the libgit2 library: {}", r);
+        let git_error = git_error_last();
+        let mut error_msg: *mut c_char = ptr::null_mut();
+        if !git_error.is_null() {
+            error_msg = (*git_error).message;
+        }
+        if !error_msg.is_null() {
+            assert!(
+                r >= 0,
+                "couldn't initialize the libgit2 library: {}, error: {}",
+                r,
+                CStr::from_ptr(error_msg).to_string_lossy()
+            );
+        } else {
+            assert!(r >= 0, "couldn't initialize the libgit2 library: {}", r);
+        }
```

Comment:
Could this perhaps be structured to check the return code, and then if it's negative fetch the error message and print it?

Reply:
Sure. Will do.

Reply:
I've commit https://github.com/rust-lang/git2-rs/pull/517/commits/83c5b25ba429772c77bf5e9f7395095624f30fa5 for your suggested comment.